### PR TITLE
Pytest merge into development

### PR DESCRIPTION
**Pull request by SafetImamovic**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -39,10 +39,10 @@
...
 
 
-@app.on_event("startup")
-def on_startup():
-    """On server startup, seed the database."""
-    db = SessionLocal()
-    uc = UtilityController(db);
-    seed(uc)
-    db.close()
+# @app.on_event("startup")
```
Comment (stankovicdejan) [inline main.py]: Is there any specific reason for leaving this commented code here?

File: tests/test_users.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        "username": "testuser2",
+        "email": "testuser2@example.com",
+        "age": 25,
+        "user_type_id": create_user_type['user_type_id'],
+        "password": "securepassword"
+    }
+
+    response = client.post(prefix + "/users/", json=user_data)
+
+    assert response.status_code == 200
```
Comment (stankovicdejan) [inline tests/test_users.py]: When some record is added to the db through the POST request it is commonly expected to have 201 - created HTTP status code, so maybe do it in both API and test and do the same for each post request and matching test

**Review by stankovicdejan — CHANGES_REQUESTED** [review]
The general approach is good, but I would suggest to implement some additional tests that should end up in some 'error' status codes, just to trigger validation that is already implemented. For example, try to send a post request with missing required field, or with a wrong value passed which cannot be accepted. In those cases you will not expect 200 http status code, but some 4xx code, like 404, 409, 422, etc. Testing just 'happy flow' is not enough and we should be sure that our tests are covering all the non valid cases (or as many of them as possible) so whatever the enduser tried to send our application can handle it.
Also, as mentioned below, 201 is a common status code for creating records / adding resources, so try to adjust that as well.

File: main.py (hunk 1 above)
Comment (SafetImamovic) [reply main.py]: Yes in case of needing to re-seed the database again on startup, but i can remove it.

**Review by stankovicdejan — APPROVED** [review]

**Pull request merged**